src/provider/inMemoryProvider.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  Address,
+  Message,
+  MessageComputer,
+  Transaction,
+  UserSecretKey,
+  UserSigner
+} from '@multiversx/sdk-core/out';
+import { IDAppProviderAccount } from '@multiversx/sdk-dapp-utils/out';
+import { IProvider } from 'types/sdkCoreTypes';
+import { signTransactions } from '@multiversx/sdk-dapp-core/out/core/providers/strategies/helpers/signTransactions/signTransactions';
+import { LoginModal } from './LoginModal';
+
+const ADDRESS_KEY = 'address';
+
+const notInitializedError = (caller: string) => () => {
+  throw new Error(`Unable to perform ${caller}, Provider not initialized`);
+};
+
+let privateKey = '';
+
+export class InMemoryProvider implements IProvider {
+  private modal = LoginModal.getInstance();
+  private _account: IDAppProviderAccount = {
+    address: sessionStorage.getItem(ADDRESS_KEY) || ''
+  };
+
+  constructor(address?: string) {
+    if (address) {
+      this.setAccount({
+        address
+      });
+    }
+  }
+
+  isInitialized = () => Boolean(this._account.address);
```

Comment:
Use regular functions in classes always, not lambdas

public isInitialized(): boolean {
   return Boolean(this._account.address);;
}